src/utils/hooks/useFetch.js

```diff
@@ -0,0 +1,52 @@
+import { useState, useEffect } from 'react'
+import { API_BASE_URL } from '../constants'
+import { useLatestAPI } from './useLatestAPI'
+
+const useFetch = (searchParams) => {
```

Comment:
This is really good, you abstracted the fetch which reduces the duplicated code by using it in the other hooks, this makes the code easier to read and maintain